_includes/utils-table.html

```diff
@@ -11,9 +11,15 @@
     <td class="util-class-cell">
       <code>.{{ util.title }}</code>
     </td>
+    {% if util.util_type contains "text-" or util.util_type contains "font-") %}
+    <td class="util-value-cell">
+      {% highlight html %}{{ util }}{% endhighlight %}
+    </td>
+    {% else %}
```

Comment:
We could avoid having to use this conditional if we use `<span>` instead of `<p>` tags for the examples.

Reply:
Ok, I was so excited about this idea and tried it... All elements were either in `p`, `div` or `span` tags, so all I did is remove the conditional.

- `divs` work just like we'd want them to
- `span` elements seem to get wrapped up in a `p` tag, maybe Jekyll does that? Or am I missing something?

The borders are the only problems here:
```
 * @example
  * <span class="s-border s-border-width-#{$width}"></span>
  */
```
![image](https://user-images.githubusercontent.com/25447342/116285053-fe40a500-a74a-11eb-8842-826af1a75d78.png)

Reply:
Sorry, maybe I wasn't super clear about my above comment. I was just saying that you don't need the `if util.util_type contains "text-" or util.util_type contains "font-")` branch of this conditional if you use `<span>` instead of `<p>` tags for your utility class examples, because then we can just assume that every example will be wrapped in a `<p>` tag and remove it using the existing filter:

```
{% highlight html %}{{ util | remove: '<p>' | remove: '</p>' }}{% endhighlight %}
```

I'm not positive, but I think that the Markdown parser used by Jekyll will wrap any markdown text (like the text in our `@example`s) in a `<p>` tag _unless_ the markdown uses a `<p>` tag as a top-level element explicitly.

Reply:
Ooh, don't be sorry. I had it backwards, I think I see what you're saying/is happening now!!

Ok here is the problem I keep running into/the reason I believe I originally got myself down this path of writing the conditional - for the utils that I want to show examples of with a paragraph tag, I **do** want both `p` tags, so the "remove" leaves me with something like this:

![image](https://user-images.githubusercontent.com/25447342/116317555-b59be280-a770-11eb-927b-bed78db337d0.png)

... which is bugging me that the closing tag is missing. Maybe I need to get over it!?

I could make them all spans in the `@example`, but for text-align utils, the rule then isn't applied since span is inline:

![image](https://user-images.githubusercontent.com/25447342/116317810-175c4c80-a771-11eb-9402-74381ba4d13d.png)

I feel I've made this way harder than it needs to be for my brain so thanks for reading.

Reply:
Got it - that makes sense. I can see why you'd want to use a `<p>` tag in this case. Well, some other options:

- Could use another block element like `<div>`
- Could get a little tricky and do something like:

```
{{ util | replace_first: '<p><', '<' | replace_first: '></p><, '>' }}
```

which would only only strip out the surrounding `<p>` tag if it contains a child tag (but not in your case above because there is no child tag, just text content).